Add info to login to a specific cluster (#3207)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.1/tutorials/login-cluster/index.md

This procedure requires the following configurations and background:

- A Konvoy cluster with [Kommander installed](https://docs.d2iq.com/ksphere/kommander/1.1/install/).
- An [Identity Provider](https://docs.d2iq.com/ksphere/kommander/1.1/tutorials/authorize-all-users/) (this tutorial uses GitHub's identity provider).
- A configured [group](https://docs.d2iq.com/ksphere/kommander/1.1/operations/identity-providers/) in Kommander.
- At least one user in that group.

## Access a cluster

If your group can access a cluster managed by Kommander, you can connect to that cluster from the Kommander landing page.

In this example, the Kommander cluster has two managed clusters on Azure but, as a user, you can only access the management cluster.

1. Go to the landing page, provided by an administrator, and select **Generate kubectl Token**.

![Konvoy Landing Page](/ksphere/kommander/1.1/img/konvoy-landing-page.png)

2. Select the cluster you want to log into.

![Generate Kubernetes Token](/ksphere/kommander/1.1/img/generate-kubernetes-token.png)

3. As a user having access to the management cluster, select the `Main` cluster and log in using the identity provider available (GitHub).

> ![Login GitHub](/ksphere/kommander/1.1/img/login-identity-provider.png)

Reviewer: Why is this specific to GitHub?

The user interface provides instructions to attach `kubectl` to the cluster:

![Connection Instructions](/ksphere/kommander/1.1/img/kubectl-connection-instructions.png)

Once you will have followed these instructions, your local `kubectl` will be able to communicate with the cluster.
Depending on your rights, you will be able to view and edit different kind of api-resources.

## Switch from a cluster to another

> If you wish to login to another cluster that is part of your Kommander infrastructure, you will have to follow the steps described above with another cluster.

Reviewer 2: ```suggestion
To log in to another cluster, that is part of your Kommander infrastructure, use the [Access a Cluster](#access_a_cluster) procedure for the new cluster.
```

> Once `kubectl` is attached to this new cluster, you can switch between clusters using [contexts](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#context). Open-source tools such as [`kubectx`](https://github.com/ahmetb/kubectx) are available to make this operation faster.

Reviewer 2: ```suggestion
Once `kubectl` is on this new cluster, you can switch between clusters using [contexts](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#context). Open-source tools such as [`kubectx`](https://github.com/ahmetb/kubectx) can make this operation faster.
```

> ### Messages when attached to a cluster that you do not have access to

Reviewer 2: ```suggestion
### Messages when attached to a cluster that you do not have access
```

> The instructions described above actually work for any clusters (management and attached ones) even if you do not have access to it.

Reviewer 2: ```suggestion
The [Access a Cluster](#access_a_cluster) instructions work for any clusters (management and attached ones) even if you do not have access to them.
```

> The difference will be that, once you have attached the cluster, none of the `kubectl` commands will succeed as the user does not have access:

Reviewer 2: ```suggestion
The difference is that, once you have attached the cluster, none of the `kubectl` commands will succeed as the user does not have access:
```
